src/hash.cr

```diff
@@ -452,6 +470,13 @@ class Hash(K, V)
     hash
   end
 
+  # Returns a new hash with the keys and values of this hash and *other* combined. 
```

Comment:
This is not true. It does not return a new hash, `merge` does. `merge!` modifies the receiver.